source/extensions/filters/http/cache/cache_filter.cc

```diff
@@ -104,8 +104,29 @@ void CacheFilter::onHeaders(LookupResult&& result) {
       state_ = GetHeadersState::GetHeadersResultUnusable;
     }
     return;
-  case CacheEntryStatus::SatisfiableRange: // TODO(#10132): break response content to the ranges
-                                           // requested.
+  case CacheEntryStatus::NotSatisfiableRange:
+    result.headers_->setStatus(static_cast<uint64_t>(Http::Code::RangeNotSatisfiable));
+    result.headers_->setContentLength(0);
+    result.headers_->addCopy(Http::Headers::get().ContentRange,
+                             absl::StrCat("bytes */", result.content_length_));
+    result.content_length_ = 0;
```

Comment:
I'm not 100% sure of the intended logic (I would guess it's not supposed to be a fallthrough, but I suspect I might be missing something), but there should be a break/return/ABSL_FALLTHROUGH_INTENDED here. (If it is a fallthrough, it's worth a comment).

Reply:
I wasn't aware of the ABSL_FALLTHROUGH_INTENDED.
My idea here was to prepare the response for either `::NotSatisfiableRange` and `::SatisfiableRange`, but let `::Ok` finish the response. This way we don't need to replicate code, and future changes would be in only one place.

I agree that treating `::NotSatisfiableRange` alongside responses that have content might be confusing though.
Do you think I should resolve this one separately instead of delegating to `::Ok`?

Reply:
I think I see: `NotSatisfiableRange` is supposed to fall through to `SatisfiableRange`, I'm guessing skip its `if (size==1)` block, and then further fall through into the `Ok` case? I see how it would be pretty clean and satisfying when you're fully comprehending the flow of the code in this file, but I think for an outsider it has subtlety that pretty substantially interferes with comprehension. Having `SatisfiableRange` use `result.content_length_` to signal into `Ok` also seems a little subtle, although I'm less sure how I feel on that.

I'm thinking it might work nicely to have the contents of the `Ok` case broken out into a helper function(s) that identify a concept(s) that both the `Ok` case and these other cases are doing, and then have the other cases call those without any fallthrough. Beyond simplifying the fallthrough stuff, I think it might make the individual cases more readable, by reading as a sequence of a few easy-to-understand calls of doThisPartOfTheProcess(), doThisOtherThing().